apps/web/pages/settings/my-account/appearance.tsx

```diff
@@ -37,21 +37,35 @@ const SkeletonLoader = () => {
 const AppearanceView = () => {
   const { t } = useLocale();
 
+  const utils = trpc.useContext();
   const { data: user, isLoading } = trpc.viewer.me.useQuery();
+  const formMethods = useForm();
+  const {
+    formState: { isSubmitting },
+  } = formMethods;
+
   const mutation = trpc.viewer.updateProfile.useMutation({
-    onSuccess: () => {
+    onSuccess: async () => {
+      await utils.viewer.me.invalidate();
       showToast(t("settings_updated_successfully"), "success");
     },
     onError: () => {
       showToast(t("error_updating_settings"), "error");
     },
   });
 
-  const formMethods = useForm();
-
   if (isLoading) return <SkeletonLoader />;
 
-  if (user)
+  const fields = formMethods.watch();
+
+  if (user) {
+    const isDisabled =
```

Comment:
in /profile we used isDirty property https://github.com/calcom/cal.com/blob/main/apps/web/pages/settings/my-account/profile.tsx#L199. but here isDirty behaviour is a bit weird it should  return true when form value change but it is only work fine in case of theme else all fields the value doesn't change. I have tried passing defaultValues in form or  resetting with keepDirtyValues true but it is not working for some reason